src/database-layer/db.ts

```diff
@@ -619,18 +630,37 @@ if (maybePg != null) {
 				}
 			}
 
-			public override async getTxLevelLock(namespaceKey: string, key: number) {
+			public override async getTxLevelLock(
+				namespaceKey: string,
+				key: number,
+				blocking: boolean = true,
+			) {
 				validateTransactionLockParameter(key, 'key');
 				const namespaceId = transactionLockNamespaceMap[namespaceKey];
 				if (namespaceId == null) {
 					throw new Error(
 						`Transaction lock namespace ${namespaceKey} not registered.`,
 					);
 				}
-				await this.executeSql(`SELECT pg_advisory_xact_lock($1, $2);`, [
-					namespaceId,
-					key,
-				]);
+				try {
+					if (blocking) {
+						const result = await this.executeSql(
+							`SELECT pg_advisory_xact_lock($1, $2);`,
+							[namespaceId, key],
+						);
+						return result.rows?.[0]?.['pg_advisory_xact_lock'] === '';
```

Comment:
If the statement returns at all that means it has to be successful right? And since the return type of `pg_advisory_xact_lock` is `void` we shouldn't really be looking at the contents
```suggestion
						return true;
```

Reply:
Fair point, will do so.